autocorpus/table.py

```diff
@@ -6,565 +6,583 @@
-                cur_superrow = next(i for i in row if i not in ("", "None"))
+    """
+    if any(char.isdigit() for char in s):
+        return False
+    return True
+
+
+def __table2json(
+    table_2d,
+    header_idx,
+    subheader_idx,
+    superrow_idx,
+    table_num,
+    title,
+    footer,
+    caption,
+):
+    """Transform tables from nested lists to JSON.
+
+    Args:
+        table_2d (list): nested list tables
+        header_idx (list): list of header indices
+        subheader_idx (list): list of subheader indices
+        superrow_idx (list): list of superrow indices
+        table_num (int): table number
+        title (str): table title
+        footer (str): table footer
+        caption (str): table caption
+
+    Returns:
+        tables (list): tables in JSON format
+
+    """
+    tables = []
+    sections = []
+    cur_table = {}
+    cur_section = {}
+
+    pre_header = []
+    pre_superrow = None
+    cur_header = ""
+    cur_superrow = ""
+    for row_idx, row in enumerate(table_2d):
+        if not any([i for i in row if i not in ["", "None"]]):
+            continue
+        if row_idx in header_idx:
+            cur_header = [
+                table_2d[i] for i in next(i for i in subheader_idx if row_idx in i)
+            ]
+        elif row_idx in superrow_idx:
+            cur_superrow = next(i for i in row if i not in ("", "None"))
+        else:
+            if cur_header != pre_header:
+                sections = []
+                pre_superrow = None
+                cur_table = {
+                    "identifier": str(table_num + 1),
+                    "title": title,
+                    "caption": caption,
+                    "columns": cur_header,
+                    "section": sections,
+                    "footer": footer,
+                }
+                tables.append(cur_table)
+            elif cur_header == pre_header:
+                cur_table["section"] = sections
+            if cur_superrow != pre_superrow:
+                cur_section = {"section_name": cur_superrow, "results": [row]}
+                sections.append(cur_section)
+            elif cur_superrow == pre_superrow:
+                cur_section["results"].append(row)
+
+            pre_header = cur_header
+            pre_superrow = cur_superrow
+
+    if len(tables) > 1:
+        for table_idx, table in enumerate(tables):
+            table["identifier"] += f".{table_idx + 1}"
+    return tables
+
+
+def __format_table_bioc(table_json, tableIdentifier, file_path):
+    bioc_format = {
+        "source": "Auto-CORPus (tables)",
+        "date": f"{datetime.today().strftime('%Y%m%d')}",
+        "key": "autocorpus_tables.key",
+        "infons": {},
+        "documents": [],
+    }
+    for table in table_json["tables"]:
+        if "." in table["identifier"] and tableIdentifier:
+            table_identifier = (
+                tableIdentifier + "_" + table["identifier"].split(".")[-1]
```

Comment:
Camelcase is a bit weird for variable names in Python anyway, but having two vars with such similar names is pretty confusing. Does `tableIdentifier` actually need to be a separate variable or could we just have the one?